### ISSUE TITLE
On Linux either the mouse pointer is not visible or not changing 

### ISSUE THREAD

**Issue by hammadmajid** [zed-industries/zed#14862]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

On Linux, the mouse pointer either remains invisible or fails to change to other pointers like select or resize. Please see the recording below for a demonstration.

**Observation/Reproduction**
- The mouse pointer never changes for me.
- The mouse pointer's invisibility appears to be random, I didn't notice any pattern every time pointer went invisible.

Possible similar issue: #13258 ?

### Environment

Zed: v0.144.4 (Zed)
OS: Linux Wayland arch unknown
Memory: 7.6 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

[Screencast from 2024-07-20 15-53-00.webm](https://github.com/user-attachments/assets/0600d5ac-15f8-4218-ae19-0a618a8a7177)

### If applicable, attach your Zed.log file to this issue.

Found relevant logs related to Wayland cursor. Almost 80% of log file is this
<details><summary>Zed.log</summary><pre>

2024-07-20T16:02:44.46382699+05:00 [WARN] Linux: Wayland: Unable to get default cursor too!
2024-07-20T16:02:47.480992442+05:00 [WARN] Linux: Wayland: Unable to get default cursor too!
2024-07-20T16:02:53.972211982+05:00 [WARN] Linux: Wayland: Unable to get default cursor too!
2024-07-20T16:02:54.862928516+05:00 [WARN] Linux: Wayland: Unable to get default cursor too!
2024-07-20T16:02:55.435338194+05:00 [WARN] Linux: Wayland: Unable to get default cursor too!
2024-07-20T16:02:55.711079273+05:00 [WARN] Linux: Wayland: Unable to get default cursor too!
2024-07-20T16:03:08.073379479+05:00 [WARN] Linux: Wayland: Unable to get default cursor too!
</pre></details>

**Comment by ukizet** [comment]
I have the same issue (nixos)

**Comment by apricotbucket28** [comment]
Hmm... it's unable to find any cursors in the cursor theme.

What's the output of `gdbus call -e -d org.freedesktop.portal.Desktop -o /org/freedesktop/portal/desktop -m org.freedesktop.portal.Settings.Read '"org.gnome.desktop.interface"' '"cursor-theme"'`?

**Comment by hammadmajid** [comment]
> What's the output of `gdbus call -e -d org.freedesktop.portal.Desktop -o /org/freedesktop/portal/desktop -m org.freedesktop.portal.Settings.Read '"org.gnome.desktop.interface"' '"cursor-theme"'`?

```
(<<'breeze_cursors'>>,)
```

**Comment by hammadmajid** [comment]
I didn't change anything in Gnome tweaks as cursor was already set to `Adwaita (default)` but I did run this command
```
gsettings set org.gnome.desktop.interface cursor-theme 'Adwaita'
```
And that seems to fix the issue I guess 

[Screencast from 2024-07-22 17-33-38.webm](https://github.com/user-attachments/assets/729b9694-c519-465a-bf9b-090ca234da39)

**Comment by megatux** [comment]
Same issue here, Wayland Nobara Linux w/breeze_cursors, changing it back to 'Adwaita' **fixed** the issue.

For me, the issue was about not _updating_ the cursor when entering inside the editor window. The former cursor icon was kept, most of the times, the resize-window icon.

**Comment by apricotbucket28** [comment]
The issue here is that there isn't a standard way to read cursor settings on Wayland, unlike X11. That's why Zed tries to use the cursor-shape-v1 protocol when available (i.e. KDE 6 and most other compositors), which solves the problem.

Unfortunately Gnome/Mutter doesn't support the protocol so instead it relies on reading the cursor theme and size from gsettings, and in these cases, it seems that said values are outdated and the themes don't exist anymore in the system.

The only other way I could see to fix this would be to add GTK as a dependency or to read GTK settings to find the desired cursor theme, but neither are ideal.

**Comment by mgsloan** [comment]
This is now fixed in #32820 and will be included in v0.191 release (probably this coming wednesday).  Adwaita theme worked because there was even a comment in the source saying the cursor icon names were based on looking at adwaita.  However, there seem to be at least 2-3 names for most cursor icons.  That PR adds those among other things.
